edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ModuleIdentityLifecycleManager.cs

```diff
@@ -27,7 +29,7 @@ public ModuleIdentityLifecycleManager(IIdentityManager identityManager, ModuleId
         public async Task<IImmutableDictionary<string, IModuleIdentity>> GetModuleIdentitiesAsync(ModuleSet desired, ModuleSet current)
```

Comment:
IMO method that returns empty list when desired and current doesn't have any differences is somewhat unexpected behavior of the method with a name `GetModuleIdentitiesAsync`. Shouldn't we find a root cause of why it was done in this manner?

Reply:
The internal function calls edgelet to list identities every time it is called.  This code was put in so we could shortcut this, because if there's no diff, there will be no changes to modules.